hardhat/contracts/BrainTrustFeeConverterContract.sol

```diff
@@ -19,18 +19,16 @@ contract BrainTrustFeeConverterContract {
     poolFee = _poolFee;
   }
 
-  function swapExactInputSingle(uint256 amountIn, uint256 amountOutMin) external returns (uint256 amountOut) {
+  function swapExactInputSingle(uint256 amountIn, uint256 amountOutMin, uint256 deadline) external returns (uint256 amountOut) {
     TransferHelper.safeTransferFrom(USDC, msg.sender, address(this), amountIn);
     TransferHelper.safeApprove(USDC, address(swapRouter), amountIn);
 
-    // Naively set amountOutMinimum to 0. In production, use an oracle or other data source to choose a safer value for amountOutMinimum.
-    // We also set the sqrtPriceLimitx96 to be 0 to ensure we swap our exact input amount.
     ISwapRouter.ExactInputSingleParams memory params = ISwapRouter.ExactInputSingleParams({
       tokenIn: USDC,
       tokenOut: BTRST,
       fee: poolFee,
       recipient: TreasuryAddress,
-      deadline: block.timestamp,
+      deadline: block.timestamp + deadline,
```

Comment:
this won't fix the issue.

block.timestamp + deadline will set the deadline to 20mins in the future of whenever the transaction is processed on-chain. this means that if someone tries to delay the transaction and process it 1 hour later, it will still get accepted as the deadline is based on transaction processing time.

we want the deadline based on transaction creation time, ie, when the user first signs the transaction in their metamask, based on the block.timestamp at signing time not at processing time.

So, in the frontend we need something like:
```
const deadline = Math.floor(Date.now() / 1000) + 60 * 10
```

and on the smart contracts, just:
```
deadline: deadline
```